### DISCUSSION TITLE
SupportBrowserHistory Query String alias bug

### DISCUSSION THREAD

**Discussion by Lupennat** [livewire/livewire#4475] [7. Bugs (archived)] [answered]
in the constructor of the class, the callback of the `component.hydrate.initial` event has a query string value assignment bug.

```php
$component->$property = $decoded === null ? $fromQueryString : $decoded;
````

`$property` is the alias instead of the original property.

properties are retrieved by the method

```php
if (!$properties = $this->getQueryParamsFromComponentProperties($component)->keys()) {
     return;
}
```

this method returns an array where the keys are the aliases, all references to the original property name are lost.

**Comment by joshhanley** [comment, marked as answer]
@Lupennat thanks for reporting! Hmm ah yeah that's not right. Was there anything that you did that triggered this issue that we can use to replicate it and write a test for it?

 @amiranagram if you get the chance, are you able to have a look at this?

**Reply by amiranagram** [reply]
Yeah, I'll give it a look. @joshhanley

**Reply by amiranagram** [reply]
Wasn't this aleady fixed in #4362? @joshhanley

**Reply by Lupennat** [reply]
@amiranagram yes you are right, it was fixed in #4362 ...

@joshhanley maybe someone should create a new tag because in the latest 2.8.2 the bug is still there

**Reply by joshhanley** [reply]
@amiranagram sorry yes it was 🤦‍♂️ thanks for checking! Only just back from holidays and had forgotten about that 😬

@Lupennat yep, we will be tagging it soon, we just didn't want to do it over the holidays as there is new features in the release too so wanted to make sure there is someone available to support them 🙂

**Reply by amiranagram** [reply]
No problem @joshhanley. Anything query strings, feel free to ping me. I've done a few PRs and I'm very familiar with URL queries in Livewire.

**Reply by joshhanley** [reply]
@amiranagram ok great! Thanks for your help, much appreciated 🙂

**Reply by Lupennat** [reply]
@joshhanley thanks, great!
